course/src/isometry.cpp

```diff
@@ -1,18 +1,110 @@
 /*
  * Isometry library
  * Copyright 2020 Ekumen Inc.
- * Author: Agustin Alba Chicar, 2019
- * Author: Gerardo Puga, 2020
+ * Author: Alexis Pojomovsky, 2020
  */
 
+#include <cmath>
+#include <iomanip>
+
 #include <isometry/isometry.hpp>
 
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+Isometry::Isometry() : rotation_{Matrix3::kZero}, translation_{Vector3()} {}
+
+Isometry::Isometry(const Vector3& translation, const Matrix3& rotation)
+    : rotation_{rotation}, translation_{translation} {}
+
+Isometry::Isometry(const Isometry& obj) = default;
+
+Isometry Isometry::fromTranslation(const Vector3& vector) {
+  return Isometry{vector, Matrix3::kIdentity};
+}
+
+Isometry Isometry::rotateAround(const Vector3& vector, const double radians) {
+  const double ux = vector[0] / vector.norm();
+  const double uy = vector[1] / vector.norm();
+  const double uz = vector[2] / vector.norm();
+  const double cosAngle = cos(radians);
+  const double sinAngle = sin(radians);
+  Matrix3 rotationMatrix =
+      Matrix3(cosAngle + std::pow(ux, 2.) * (1. - cosAngle),
+              ux * uy * (1. - cosAngle) - uz * sinAngle,
+              ux * uz * (1. - cosAngle) + uy * sinAngle,
+              uy * ux * (1. - cosAngle) + uz * sinAngle,
+              cosAngle + std::pow(uy, 2.) * (1 - cosAngle),
+              uy * uz * (1. - cosAngle) - ux * sinAngle,
+              uz * ux * (1. - cosAngle) - uy * sinAngle,
+              uz * uy * (1. - cosAngle) + ux * sinAngle,
+              cosAngle + std::pow(uz, 2.) * (1. - cosAngle));
+  return Isometry{Vector3(), rotationMatrix};
+}
+
+Isometry Isometry::fromEulerAngles(const double roll, const double pitch,
+                                   const double yaw) {
+  return Isometry::rotateAround(Vector3::kUnitX, roll) *
+         Isometry::rotateAround(Vector3::kUnitY, pitch) *
+         Isometry::rotateAround(Vector3::kUnitZ, yaw);
```

Comment:
nit: `rotateAround` is member method. You don't need to add the namespace `Isometry::`